Add backup scripts

kamfonik opened the pull request:
Further parameterizes the backup scripts and adds them to this repo.

Requires these parameters to be updated:
quickstack::params::backups_script_controller: 'puppet:///modules/quickstack/controller_backup.sh'
quickstack::params::backups_script_compute: 'puppet:///modules/quickstack/compute_backup.sh'

I also added these additional parameters:
quickstack::params::backups_enabled: 'true'
quickstack::params::backups_verbose: 'true'
quickstack::params::backups_keep_days: '3'

kamfonik committed:
add parameterized backup scripts and parameters

kamfonik committed:
add logrotate for backups logs

kamfonik committed:
add verbose flag parameter

kamfonik committed:
rsync needs /var/log directory to be made beforehand

kamfonik committed:
typo fix

fix verbosity if statement

update staging-area collector key to compute-203

fix typo in controller_backup.sh

kamfonik commented:
This has been tested in the virtual staging area, but not on physical nodes.

rahulait commented:
@kamfonik Should we close this and merge to production? There are some conflicts which might need to be fixed, let us know when you are done, me and Rado can then review and merge this one.

radonm commented:
Can you add a check for a variable from the hiera file and only include backup class if that variable has a certain value? That way we can include/exclude backups as needed.
There is a conflict with rsync package definition - it's in backups and now (cluster pull request) in controller_common, since backup would be optional I left in controller common.

kamfonik committed:
remove rsync package from backups now that it is defined in controller_common

kamfonik commented:
@radonm We should probably also define the rsync package in compute_common, then.  I will make the change.

kamfonik committed:
add enable switch for backups

update params.pp

fix backup enable

kamfonik committed:
convert backups crontab entry to cron.d script

remove unnecessary staging area public key

add missing parameter to compute_common.pp

remove old backups crontab entry

remove commented out code

rahulait commented on quickstack/manifests/backups.pp:
```diff
@@ -101,20 +99,26 @@
-      environment => "MAILTO=${cron_email}",
-      hour        => $cron_hour,
-      minute      => $cron_min,
+        ensure   => absent,
```
We want the 'backup_cron' to be absent or present? Is this by mistake?

kamfonik replied:
It replaces the crontab entry backup_cron and with the file backup_cron (defined just above) which is a cron.d script.  We need to make sure the crontab entry gets deleted otherwise the scripts will run twice.  

Once all the nodes get this change, we can take out the code removing the crontab entry.

rahulait commented:
+1 looks good to me except the one comment. Also, let's try to follow the arrow alignment:-
https://docs.puppet.com/guides/style_guide.html#arrow-alignment

Else, the code will look messy after sometime. :)

radonm commented:
+1
